pkg/polaris/polarisMutator.go

```diff
@@ -43,7 +43,12 @@ func GetPolarisComponents(baseUrl string, polaris Polaris) (map[string]runtime.O
 	content = strings.ReplaceAll(content, "${ENVIRONMENT_NAME}", polaris.Namespace)
 	content = strings.ReplaceAll(content, "${POLARIS_ROOT_DOMAIN}", polaris.EnvironmentDNS)
 	content = strings.ReplaceAll(content, "${IMAGE_PULL_SECRETS}", polaris.ImagePullSecrets)
-	content = strings.ReplaceAll(content, "${DOWNLOAD_SERVER_PV_SIZE}", polaris.PolarisSpec.DownloadServerDetails.Storage.StorageSize)
+
+	if len(polaris.PolarisSpec.DownloadServerDetails.Storage.StorageSize) != 0 {
```

Comment:
polaris.PolarisSpec.DownloadServerDetails.Storage.StorageSize will always have value because of GetPolarisDefault() so you don't need an if statement here.

Reply:
Gotcha. Removing the condition..